### ISSUE TITLE
Install error during setup.sh script

### ISSUE THREAD

**Issue by sjd-xlnx** [CESNET/netopeer2#977]
Hi, I'm building from github master branch. During "make install" I get an error. I'm trying to build as Debug and installing in local directory as this for is a development / test environment / experiment.

cmake and make already completed successfully.

```
make install
[ 56%] Built target serverobj
[ 69%] Built target netopeer2-server
[100%] Built target netopeer2-cli
Install the project...
-- Install configuration: "Debug"
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/iana-crypt-hash@2014-08-06.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-crypto-types@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-datastores@2017-08-17.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-interfaces@2018-02-20.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-ip@2018-02-22.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-keystore@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-netconf-acm@2018-02-14.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-netconf-monitoring@2010-10-04.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-netconf-nmda@2019-01-07.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-netconf-server@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-netconf@2013-09-29.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-network-instance@2019-01-21.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-restconf@2017-01-26.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-ssh-common@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-ssh-server@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-subscribed-notifications@2019-09-09.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-tcp-client@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-tcp-common@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-tcp-server@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-tls-common@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-tls-server@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-truststore@2019-07-02.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-x509-cert-to-name@2014-12-10.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-yang-patch@2017-02-22.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-yang-push@2019-09-09.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/ietf-yang-schema-mount@2019-01-14.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/nc-notifications@2008-07-14.yang
-- Up-to-date: /home/stevend/.local/share/yang/modules/netopeer2/notifications@2008-07-14.yang
-- Installing: /home/stevend/.local/bin/netopeer2-server
-- Removed runtime path from "/home/stevend/.local/bin/netopeer2-server"
-- Installing missing sysrepo modules...
[ERR]: Invalid context for LYB data parsing, missing module "ietf-yang-push@2019-09-09".
[ERR]: Failed to update data for the new context.
sysrepoctl error: Failed to apply changes (Operation failed)
For more details you may try to increase the verbosity up to "-v3".
CMake Error at cmake_install.cmake:74 (message):
   scripts/setup.sh failed: 1
```

Everything seems fine until it tries to sysrepoctl install the "ietf-yang-push@2019-09-09" file. The attempted install fails, and corrupts the sysrepo repository (e.g. most other sysrepoctl or sysrepocfg command fail afterwards).

I need to delete the sysrepo repository afterwards to remove the corruption.

If I comment out the line "ietf-yang-push@2019-09-09" from the scripts/setup.sh then make install succeeds.

I also tried to manually install all the files that are normally installed by scripts/setup.sh, they all work except for the last in the list - the "ietf-yang-push@2019-09-09".

It seems like there is something wrong with this file. Although it does pass pyang --lint. I don't know enough about Yang to manually check the file.

Any idea of the problem?

**Comment by michalvasko** [comment]
I am not sure what the problem may be but you should definitely try using `devel` branch instead of the `master`, there are lots of improvements including some in the build itself.

**Comment by sjd-xlnx** [comment]
Thanks for quick response Michal.

I updated to "devel" branch (netopeer2, sysrepo and libnetopeer2). This didn't immediately fix the problem, but I got a new error relating to shared memory conflict.

`[ERR]: Shared memory version mismatch (6, expected 7), remove the SHM to fix.`

Digging deeper I find that this issue can occur when changing sysrepo version (see https://github.com/sysrepo/sysrepo/issues/2308).

The suggested remedy is to `make shm_clean` which I did, and now it builds & installs correctly.
